tests/lambda_endpoints/test_batch_starter.py

```diff
@@ -238,6 +238,27 @@ def test_pre_processing_dependency(session):
     assert downstream_dependency[0]["descriptor"] == "norm-mago"
 
 
+def test_duplicate_dependencies(session):
+    """Tests the unique constraint in the dependency table."""
+    _populate_dependency_table(session)
+    _populate_dependency_table(session)
```

Comment:
Just a note for future selves that this might cause an issue if/when we make a unique constraint on the table. I think this is still a good test and we'll just need to verify that this second call raises the expected error.

Reply:
Agreed, but this test does mean that we need to understand what the consequences are if this method is called twice, which is good enough for me right now!